### ISSUE TITLE
[CT-1084] [Bug] dbt deps fails on WSL 1

### ISSUE THREAD

**Issue by unrepeat** [dbt-labs/dbt-core#5712]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

When using WSL 1 install deps failed with message:
```
PermissionError: [Errno 13] Permission denied: 'dbt_packages/dbt-utils-0.8.6' -> 'dbt_packages/dbt_utils'
```
State of the packages folder after command failed:
```
ls -l dbt_packages/
total 0
drwxrwxrwx 1 krawczuj krawczuj 4096 Jun 15 21:44 dbt-utils-0.8.6
drwxrwxrwx 1 krawczuj krawczuj 4096 Jun 15 21:44 dbt_utils
```

### Expected Behavior

dbt deps should be installed correctly

### Steps To Reproduce

1. python -m venv .venv
2. source .venv/bin/activate
3. pip install dbt-redshift
4. dbt clean
5. dbt deps

### Relevant log output

```shell
$ dbt deps
14:00:10  Running with dbt=1.2.0
14:00:12  Installing dbt-labs/dbt_utils
Traceback (most recent call last):
  File "/usr/lib/python3.8/shutil.py", line 791, in move
    os.rename(src, real_dst)
PermissionError: [Errno 13] Permission denied: 'dbt_packages/dbt-utils-0.8.6' -> 'dbt_packages/dbt_utils'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/main.py", line 129, in main
    results, succeeded = handle_and_check(args)
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/main.py", line 192, in handle_and_check
    task, res = run_from_args(parsed)
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/main.py", line 239, in run_from_args
    results = task.run()
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/task/deps.py", line 67, in run
    package.install(self.config, renderer)
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/deps/registry.py", line 74, in install
    connection_exception_retry(download_untar_fn, 5)
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/utils.py", line 616, in _connection_exception_retry
    return fn()
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/deps/registry.py", line 86, in download_and_untar
    system.untar_package(tar_path, deps_path, package_name)
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/clients/system.py", line 483, in untar_package
    dbt.clients.system.rename(downloaded_path, desired_path, force=True)
  File "/mnt/c/Users/krawczuj/repositories/.testenv/lib/python3.8/site-packages/dbt/clients/system.py", line 471, in rename
    shutil.move(from_path, to_path)
  File "/usr/lib/python3.8/shutil.py", line 807, in move
    copytree(src, real_dst, copy_function=copy_function,
  File "/usr/lib/python3.8/shutil.py", line 557, in copytree
    return _copytree(entries=entries, src=src, dst=dst, symlinks=symlinks,
  File "/usr/lib/python3.8/shutil.py", line 513, in _copytree
    raise Error(errors)
shutil.Error: [(<DirEntry '.github'>, 'dbt_packages/dbt_utils/.github', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/.github'"), (<DirEntry 'data'>, 'dbt_packages/dbt_utils/integration_tests/data', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/integration_tests/data'"), (<DirEntry 'models'>, 'dbt_packages/dbt_utils/integration_tests/models', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/integration_tests/models'"), (<DirEntry 'tests'>, 'dbt_packages/dbt_utils/integration_tests/tests', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/integration_tests/tests'"), (<DirEntry 'integration_tests'>, 'dbt_packages/dbt_utils/integration_tests', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/integration_tests'"), (<DirEntry 'macros'>, 'dbt_packages/dbt_utils/macros', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/macros'"), (<DirEntry 'tests'>, 'dbt_packages/dbt_utils/tests', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/tests'"), ('dbt_packages/dbt-utils-0.8.6', 'dbt_packages/dbt_utils', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils'")]
14:00:18  Encountered an error:
[(<DirEntry '.github'>, 'dbt_packages/dbt_utils/.github', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/.github'"), (<DirEntry 'data'>, 'dbt_packages/dbt_utils/integration_tests/data', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/integration_tests/data'"), (<DirEntry 'models'>, 'dbt_packages/dbt_utils/integration_tests/models', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/integration_tests/models'"), (<DirEntry 'tests'>, 'dbt_packages/dbt_utils/integration_tests/tests', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/integration_tests/tests'"), (<DirEntry 'integration_tests'>, 'dbt_packages/dbt_utils/integration_tests', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/integration_tests'"), (<DirEntry 'macros'>, 'dbt_packages/dbt_utils/macros', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/macros'"), (<DirEntry 'tests'>, 'dbt_packages/dbt_utils/tests', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils/tests'"), ('dbt_packages/dbt-utils-0.8.6', 'dbt_packages/dbt_utils', "[Errno 16] Device or resource busy: 'dbt_packages/dbt_utils'")]
```

### Environment

- OS:
```markdown
wsl -l -v
  NAME            STATE           VERSION
* Ubuntu-20.04    Running         1
```
- Python: 3.8
- dbt: 1.2.0

### Which database adapter are you using with dbt?

redshift

### Additional Context

packages.yml
```
packages:
  - package: dbt-labs/dbt_utils
    version: [">=0.8.0", "<0.9.0"]
  - package: calogica/dbt_expectations
    version: [">=0.5.0", "<0.6.0"]
```

**Comment by unrepeat** [comment]
also same problem with dbt-core 1.0.8 and 1.1.2

**Comment by lostmygithubaccount** [comment]
thanks for reporting @unrepeat! this seems simple enough to reproduce, but I don't have windows/WSL easily available. @jtcohen6 how should we proceed on things like this? 

quick search yielded this, which sounds similar: https://stackoverflow.com/questions/24857779/why-os-rename-sometimes-returns-error-access-is-denied-python

**Comment by jtcohen6** [comment]
Not exactly the same error, but it does remind me of: https://github.com/dbt-labs/dbt-core/issues/4372

In both cases, I believe it's a matter of dbt being unable to rename/replace files on Windows machines, because it has insufficient permissions on the file system.

We're not users of Windows/WSL ourselves. My inclination is to mark this as a "help wanted" for someone who is. This is self-contained code within `dbt-core`. It's just calling `shutil.move`, which in this case encounters a `PermissionError`:

https://github.com/dbt-labs/dbt-core/blob/49e7bdbef957c76312affb7375c3983e4a749b65/core/dbt/clients/system.py#L460-L471

**Comment by Oracen** [comment]
I have a Windows machine at home, I'd be a first-time contributor but I'm happy to take this on. I'll replicate the issue and then work on a fix

**Comment by Oracen** [comment]
I ran an initial test with a project I've been tinkering with, and my initial sense is this is a local permissions issue:
```
oracen@DESKTOP-ETEIAO1:/mnt/d/projects/stacks/stack_test$ cd data-modelling/remote_sales/
oracen@DESKTOP-ETEIAO1:/mnt/d/projects/stacks/stack_test/data-modelling/remote_sales$ poetry run dbt deps
12:24:07  Running with dbt=1.3.0
12:24:08  Installing dbt-labs/metrics
12:24:14    Installed from version 1.3.1
12:24:14    Up to date!
oracen@DESKTOP-ETEIAO1:/mnt/d/projects/stacks/stack_test/data-modelling/remote_sales$
```
This was using Poetry for the venv, and WSL2. I'll spend a bit more time tomorrow morning confirming that the setup listed by @unrepeat works on my machine, and if it does we can probably call this closed. I don't think the differences between WSL1 and 2 would make much of a difference. My hunch would be filesystem permissions due to where the local version of Python/virtualenv is installing its envs.

**Comment by Oracen** [comment]
```
oracen@DESKTOP-ETEIAO1:/mnt/d/projects/stacks/stack_test/data-modelling/remote_sales$ poetry run dbt deps               09:33:50  Running with dbt=1.3.0                                                                                        09:33:54  Installing dbt-labs/dbt_utils                                                                                 09:34:03    Installed from version 0.8.6                                                                                09:34:03    Updated version available: 0.9.2                                                                            09:34:03  Installing calogica/dbt_expectations                                                                          09:34:08    Installed from version 0.5.8                                                                                09:34:08    Updated version available: 0.7.0                                                                            09:34:08  Installing calogica/dbt_date                                                                                  09:34:10    Installed from version 0.5.7                                                                                09:34:10    Updated version available: 0.6.3                                                                            09:34:10                                                                                                                09:34:10  Updates available for packages: ['dbt-labs/dbt_utils', 'calogica/dbt_expectations', 'calogica/dbt_date']                                                                                                                              Update your versions in packages.yml, then run dbt deps                                                                 oracen@DESKTOP-ETEIAO1:/mnt/d/projects/stacks/stack_test/data-modelling/remote_sales$  
```

@jtcohen6 I think we can call this closed? It's a permissions issue, not codebase. Windows isn't the best for interoperability
